runtime/src/chplsys.c

```diff
@@ -526,103 +546,126 @@ static void getCpuInfo(int* p_numPhysCpus, int* p_numLogCpus) {
+    int physIdTmp;
+    int coreIdTmp;
 
     if (sscanf(buf, "processor : %i", &procTmp) == 1) {
       procs++;
     }
-    else if (sscanf(buf, "cpu cores : %i", &cpuCoresTmp) == 1) {
-      if (cpuCores == 0)
-        cpuCores = cpuCoresTmp;
-      else if (cpuCoresTmp != cpuCores)
-        chpl_internal_error("varying number of cpu cores");
+    else if (sscanf(buf, "physical id : %i", &physIdTmp) == 1) {
+      if (physId >= 0) {
+        add_to_cpuTab(physId, coreId);
+        coreId = -1;
+      }
+      physId = physIdTmp;
     }
-    else if (sscanf(buf, "siblings : %i", &siblingsTmp) == 1) {
-      if (siblings == 0)
-        siblings = siblingsTmp;
-      else if (siblingsTmp != siblings)
-        chpl_internal_error("varying number of siblings");
+    else if (sscanf(buf, "core id : %i", &coreIdTmp) == 1) {
+      if (coreId >= 0) {
+        add_to_cpuTab(physId, coreId);
+        physId = -1;
+      }
+      coreId = coreIdTmp;
     }
 
     while (buf[buf_len - 1] != '\n' && fgets(buf, sizeof(buf), f) != NULL)
       buf_len = strlen(buf);
   }
 
+  if (physId >= 0 || coreId >= 0) {
+    add_to_cpuTab(physId, coreId);
+  }
+
   fclose(f);
 
-  if (cpuCores == 0 && siblings == 0) {
-    // We have a limited-format /proc/cpuinfo.
-    // See if the /sys filesystem has any more information for us.
-    int threads_per_core = 0;
-    if ((f = fopen("/sys/devices/system/cpu/cpu0/topology/thread_siblings",
-                   "r")) != NULL) {
-      int c;
-      while ((c = getc(f)) != EOF) {
-        // The number of threads per core is the total number of bits
-        // set in the hex digits of the thread_siblings map.
-        //
-        // The most authoritative source, kernel.org, does not dictate
-        // the formatting of the mask.  However, the following document
-        // indicates that it is either a hex or binary bit mask.  The code
-        // below works in either case.
-        //
-        // https://www.ibm.com/support/knowledgecenter/en/linuxonibm/liaat/liaattunproctop.htm
-        //
-        // Also note that hwloc itself parses the file as a hex bitmap only.
-        if (isxdigit(c)) {
-          switch (tolower(c)) {
-          case '0':
-            break;
-          case '1':
-          case '2':
-          case '4':
-          case '8':
-            threads_per_core += 1;
-            break;
-          case '3':
-          case '5':
-          case '6':
-          case '9':
-          case 'a':
-          case 'c':
-            threads_per_core += 2;
-            break;
-          case '7':
-          case 'b':
-          case 'd':
-          case 'e':
-            threads_per_core += 3;
-            break;
-          case 'f':
-            threads_per_core += 4;
-            break;
-          }
-        } else if (c != ',' && c != '\n' && tolower(c) != 'x') {
-          // unknown file format -- don't use
-          threads_per_core = 1;
+  if ((numPUs = procs) <= 0)
+    numPUs = 1;
```

Comment:
We ought to see the `processor : <integer>` lines in every `/proc/cpuinfo` but I'm finding some pretty strange `/proc/cpuinfo` instances out there, so it's probably better to assume they might be missing even when we see the `core id` and `physical id` lines.  In that case, instead of defaulting `numPUs` to 1, it would be better to default it to whatever we end up setting `numCores` to.  That way, `numPUs` will always be at least as big as `numCores`, which I think is something we should expect.